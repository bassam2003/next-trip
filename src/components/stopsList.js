import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import { Paper, List, ListItem } from '@material-ui/core'
import { header } from '../utils/styleConstants'

const styles = {
  header,
}

export const StopsList = ({ 
  stops = [], 
  classes = {}, 
  onStopClicked = () => {} 
}) => 
  <Paper style={{ margin: '20px', textAlign: 'center' }}>
    <h1 className={classes.header}>Stops</h1>
    <List style={{ maxHeight: '500px', overflow: 'scroll' }}>
      {stops.map((stop, idx) => 
        <ListItem 
          button
          onClick={() => onStopClicked(stop)}
          value={stop.Value} 
          key={idx}
        >
          {stop.Text}
        </ListItem>
      )}
    </List>
  </Paper>

export default withStyles(styles)(StopsList)