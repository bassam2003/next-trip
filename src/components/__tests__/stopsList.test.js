import React from 'react'
import { shallow } from 'enzyme'
import { StopsList } from '../stopsList'

describe('StopsList', () => {
  it('should render without error', () => {
    const wrapper = shallow(<StopsList />)

    expect(wrapper).toMatchSnapshot()
  })
})