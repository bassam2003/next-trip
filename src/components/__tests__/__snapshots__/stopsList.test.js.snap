// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`StopsList should render without error 1`] = `ShallowWrapper {}`;
